packages/reward-root-submitter/reward_root_submitter/main.py

```diff
@@ -0,0 +1,47 @@
+import logging
+import os
+import time
+
+import schedule
+from dotenv import load_dotenv
+from web3 import Web3
+
+from .submitter import RootSubmitter
+
+load_dotenv()  # take environment variables from .env
+
+LOGLEVEL = os.environ.get("LOGLEVEL", "WARNING").upper()
+logging.basicConfig(level=LOGLEVEL)
+
+
+def run_all():
+
+    for expected_env in [
+        "ETHEREUM_NODE_URL",
+        "OWNER",
+        "OWNER_PRIVATE_KEY",
+        "REWARD_POOL_ADDRESS",
+        "REWARD_PROGRAM_OUTPUT",
+    ]:
+        if expected_env not in os.environ:
+            raise ValueError(f"Missing environment variable {expected_env}")
+
+    submitter = RootSubmitter(
+        Web3(Web3.HTTPProvider(os.getenv("ETHEREUM_NODE_URL"))),
+        os.getenv("OWNER"),
+        os.getenv("OWNER_PRIVATE_KEY"),
+        os.getenv("REWARD_POOL_ADDRESS"),
+        os.getenv("REWARD_PROGRAM_OUTPUT"),
+    )
+    # Default to one minute
+    frequency = os.getenv("SUBMIT_FREQUENCY", 60)
+    schedule.every(frequency).seconds.do(submitter.submit_all_roots)
+    schedule.run_all()
+    # Go into an infinite loop
+    while True:
+        schedule.run_pending()
```

Comment:
I guess in this case. The scheduler is for computing `reward_output`. But the merkle root submission will be scheduled inside of this repo?

Reply:
Yeah - this is for a couple of reasons. One is that the logic here may end up somewhere quite different, but also this allows airdrops & calculations outside of the regular scheduling.

This could be a chained task though, triggered when the rules are computed & stored to s3.